packages/react/src/lib/positioner/Popper.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react'
+import PopperJS from 'popper.js'
+import { Ref } from '@stardust-ui/react-component-ref'
+
+import { getPlacement, applyRtlToOffset } from './positioningHelper'
+import { PopperProps, PopperChildrenFn } from './types'
+
+/**
+ * Popper relies on the 3rd party library [Popper.js](https://github.com/FezVrasta/popper.js) for positioning.
+ */
+const Popper: React.FunctionComponent<PopperProps> = props => {
+  const {
+    align,
+    children,
+    eventsEnabled,
+    modifiers,
+    offset,
+    pointerRef,
+    position,
+    positionFixed,
+    positioningDependencies = [],
+    rtl,
+    targetRef,
+  } = props
+
+  const proposedPlacement = getPlacement({ align, position, rtl })
+
+  const popperRef = React.useRef<PopperJS>()
+  const contentRef = React.useRef<HTMLElement>(null)
+  const latestPlacement = React.useRef<PopperJS.Placement>(proposedPlacement)
+  const [computedPlacement, setComputedPlacement] = React.useState<PopperJS.Placement>(
+    proposedPlacement,
+  )
+
+  const computedModifiers: PopperJS.Modifiers = React.useMemo(
+    () =>
+      offset && {
+        offset: { offset: rtl ? applyRtlToOffset(offset, position) : offset },
+        keepTogether: { enabled: false },
+      },
+    [rtl, offset, position],
+  )
+
+  React.useEffect(
+    () => {
+      const handleUpdate = (data: PopperJS.Data) => {
+        // PopperJS performs computations that might update the computed placement: auto positioning, flipping the
+        // placement in case the popper box should be rendered at the edge of the viewport and does not fit
+        if (data.placement !== latestPlacement.current) {
+          latestPlacement.current = data.placement
+          setComputedPlacement(data.placement)
+        }
+      }
+
+      const pointerRefElement = pointerRef && pointerRef.current
+      const options: PopperJS.PopperOptions = {
+        placement: proposedPlacement,
+        eventsEnabled,
+        positionFixed,
+        modifiers: {
+          ...computedModifiers,
+          ...modifiers,
+          arrow: {
+            enabled: !!pointerRefElement,
+            element: pointerRefElement,
```

Comment:
nit: lets rename it to `pointerTarget` (and corresponding `pointingTargetRefElement`). Otherwise from the code it was unclear whether this implies element that is pointer _itself_ (like arrow), or its target, or something else

Reply:
but `pointerRef` is more aligned to current `targetRef`:

```typescript
/**
 * Ref object containing the pointer node.
 */
pointerRef?: React.RefObject<Element>

/**
 * Ref object containing the target node (the element that we're using as reference for Popper box).
 */
targetRef?: React.RefObject<Element>
```

To me it seems that having `pointerTarget` and `targetRef` is more confusing